HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc

```diff
@@ -94,102 +97,167 @@ template<typename T>
 bool
 HLTJetSortedVBFFilter<T>::hltFilter(edm::Event& event, const edm::EventSetup& setup,trigger::TriggerFilterObjectWithRefs& filterproduct) const
 {
+	using namespace std;
+	using namespace edm;
+	using namespace reco;
+	using namespace trigger;
+
+	typedef vector<T> TCollection;
+	typedef Ref<TCollection> TRef;
+
+	bool accept(false);
+	
+	Handle<TCollection> jets;
+	event.getByToken(m_theJetsToken,jets);
+	Handle<JetTagCollection> jetTags;
+
+	if (saveTags()) filterproduct.addCollectionTag(inputJets_);
+	if (jets->size()<4) return false;
+
+	const unsigned int nMax(njets_<jets->size()?njets_:jets->size());
+	vector<Jpair> sorted(nMax);
+	vector<TRef> jetRefs(nMax);
+
+	unsigned int nJet=0;
+	double value(0.0);
+
+	Particle::LorentzVector b1,b2,q1,q2;
+	if (inputJetTags_.encode()=="") {
+		for (typename TCollection::const_iterator jet=jets->begin(); (jet!=jets->end()&& nJet<nMax); ++jet) {
+			if (value_=="Pt") {
```

Comment:
would be much better to convert value_ into an enum of sorts as to avoid several string comparisons per jets